Bottom: 3f518c50d8354809224e4ab4b34e2a670885a71a
Top:    d57e3afd8921c3acb0cf93ffb59125d6e9477c5a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:52:22 +0000

hid/common/hidnogui.c: Plug leak in nogui_prompt_for and nogui_fileselect

We forgot to free() the memory returned from read_stdin_line(). There is
actually no need to strdup the answer anyway, so pass the returned value
directly and let the caller free it.

(Caught with cppcheck)


---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 9d88086..b08d90d 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -381,7 +381,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
   if (answer == NULL)
     return strdup ((default_string != NULL) ? default_string : "");
   else
-    return strdup (answer);
+    return answer;
 }
 
 /* FIXME - this could use some enhancement to actually use the other
@@ -402,7 +402,7 @@ nogui_fileselect (const char *title, const char *descr,
   if (answer == NULL)
     return (default_file != NULL) ? strdup (default_file) : NULL;
   else
-    return strdup (answer);
+    return answer;
 }
 
 static int
